### PR TITLE
chore: update bloc

### LINKED ISSUE #116: Migrate to Bloc 6.0
Bloc released some new versions which makes it easier to understand the code, we are still using v4.0.0 Fortunately the documentation has a really good guide to help with the migration which can be found [here](https://bloclibrary.dev/#/migration). The user who gets this assigned will have to first move to bloc 5.0 and then 6.0 as there are quite a few breaking changes and we dont want to miss any.

### PR DESCRIPTION
**Pull request by techno-disaster**
### Description
Updated Bloc to 6.0.1

Fixes #116 

### Flutter Channel:
- [x] I have used the Flutter Beta channel on my local machine 

### Type of Change:
**Delete irrelevant options.**

- Code

### How Has This Been Tested?
Physical Device

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged

### CODE REVIEW

**Comment by techno-disaster** [comment]
@anitab-org/qa-team can someone give this a quick test? updated dependency, so you will have to test whole apk

**Comment by techno-disaster** [comment]
ah the build fails because the flutter sdk used in github actions is 1.15. I have already updates this in #102 . We will have to merge that first

**Review by robotjellyzone — DISMISSED** [review]
These slight changes look good to me 👍 lets wait for its verification through testing for its complete review .

**Comment by techno-disaster** [comment]
@anitab-org/mentorship-flutter-maintainers can you review this. Thanks you

**Comment by techno-disaster** [comment]
@isabelcosta Can you review this?

File: lib/screens/home/pages/stats/stats_page.dart
Comment (isabelcosta) [inline lib/screens/home/pages/stats/stats_page.dart:4]: why was this removed @Techno-Disaster ?

Comment (techno-disaster) [reply lib/screens/home/pages/stats/stats_page.dart:4]: @isabelcosta thats a unused import.

**Review by isabelcosta — APPROVED** [review]

**Comment by techno-disaster** [comment]
@isabelcosta can you merge this? Offline support PR #133  also depends on this

**Comment by isabelcosta** [comment]
> @isabelcosta can you merge this? Offline support PR #133 also depends on this

Can you ask for one more review and someone to test the PR? So we follow the guidelines :) After that, I can totally merge. @Techno-Disaster

**Comment by techno-disaster** [comment]
> > @isabelcosta can you merge this? Offline support PR #133 also depends on this
> 
> Can you ask for one more review and someone to test the PR? So we follow the guidelines :) After that, I can totally merge. @Techno-Disaster

@anitab-org/coding-team can someone review this PR?

**Review by robotjellyzone — APPROVED** [review]
LGTM 👍

**Comment by isabelcosta** [comment]
@anitab-org/qa-team can someone please just run the flutter app and see if everything is working as is should?

**Review by robotjellyzone — APPROVED** [review]
All the changes were tested locally [as can be seen from the following gif] and its working fine!! good job @Techno-Disaster 👍 

<div height="50" width="50">

</div>

**Comment by robotjellyzone** [comment]
Just a small query @isabelcosta how to reduce the size of the gifs as I don't know whenever i directly drop the gif in the comment section it just take a monster form 😆 .... 😄

**Comment by garimasingh128** [comment]
> Just a small query @isabelcosta how to reduce the size of the gifs as I don't know whenever i directly drop the gif in the comment section it just take a monster form 😆 .... 😄

The markdown works in the same way as HTML does. So you can use something like this :) :
```
<div height="200" width="200">
<img src=".....">
</div>
```

**Comment by robotjellyzone** [comment]
> > Just a small query @isabelcosta how to reduce the size of the gifs as I don't know whenever i directly drop the gif in the comment section it just take a monster form 😆 .... 😄
> 
> The markdown works in the same way as HTML does. So you can use something like this :) :
> 
> ```
> <div height="200" width="200">
> 
> </div>
> ```

hmm... i know this and i have tried this many times but this didn't work for me . will it work for gif ? hmmm ok lemme try ! oh!!! tried and not working as i said already BTW thanks for your reply 😄

**Comment by techno-disaster** [comment]
@isabelcosta can we merge this now?

**Comment by techno-disaster** [comment]
Hey @isabelcosta , can we merge this now? a lot of PRs are being blocked by this.

**Pull request merged**
